common/transform/src/main/java/ai/h2o/mojos/deploy/common/transform/MojoScorer.java

```diff
@@ -64,13 +87,103 @@ public MojoScorer(
    * @return response {@link ScoreResponse}
    */
   public ScoreResponse score(ScoreRequest request) {
-    MojoFrame requestFrame = requestConverter.apply(request, pipeline.getInputFrameBuilder());
+    MojoFrame requestFrame = scoreRequestConverter
+            .apply(request, pipeline.getInputFrameBuilder());
     MojoFrame responseFrame = doScore(requestFrame);
-    ScoreResponse response = responseConverter.apply(responseFrame, request);
+    ScoreResponse response = scoreResponseConverter.apply(responseFrame, request);
     response.id(pipeline.getUuid());
+    try {
+      ShapleyType requestedShapleyType = shapleyType(request.getShapleyValuesRequested());
+      switch (requestedShapleyType) {
+        case TRANSFORMED:
+          response.setFeatureShapleyContributions(computeContribution(request));
+          break;
+        case ORIGINAL:
+          log.info(UNIMPLEMENTED_MESSAGE);
+          break;
+        default:
+          break;
+      }
+    } catch (Exception e) {
+      log.info("Failed shapley values: {}, due to: {}", request, e.getMessage());
+      log.debug(" - failure cause: ", e);
+    }
     return response;
   }
 
+  /**
+   * Method to get shapley values for an incoming request of type {@link ScoreRequest}.
+   *
+   * @param request {@link ScoreRequest}
+   * @return response {@link ContributionResponse}
+   */
+  private ContributionResponse computeContribution(ScoreRequest request) {
```

Comment:
readability: overloading only on the input type can be quite confusing if the methods contain the is different in other ways then the type being handled.

----
I'd even recommend to avoid overloading as much as possible except constructors/factories and differentiate the methods also in name.